contracts/eth/nearprover/utils/upgrade_admin.js

```diff
@@ -0,0 +1,80 @@
+const { ethers } = require('hardhat');
+const { assert, expect } = require('chai');
+const { EthLedgerSigner } = require('./eth-ledger-signer');
+const yesno = require('yesno');
+
+async function getSlotsData (provider, contractAddress, numOfSlotsToDisplay) {
+    console.log(`Contract's ${contractAddress} slots:`);
+    for (let i = 0; i < numOfSlotsToDisplay; ++i) {
+        const slotData = await provider.getStorageAt(contractAddress, i);
+        console.log(`Slot ${i}: ${slotData}`);
+    }
+}
+
+async function upgradeAdminAddressTo ({
+    provider,
+    contractAddress,
+    currentAdminAddress,
+    newAdminAddress,
+    adminAddressSlot,
+    ledgerKeyPath,
+}) {
+    const currentAdminAtSlot = ethers.BigNumber.from(await provider.getStorageAt(contractAddress, Number(adminAddressSlot))).toHexString();
+    assert.equal(
+        currentAdminAtSlot,
+        currentAdminAddress,
+        `The current admin doesn't match at the slot ${adminAddressSlot} contract ${contractAddress}`,
+    );
+
+    const adminControlledFactory = await ethers.getContractFactory('AdminControlled');
+    const adminControlled = adminControlledFactory.attach(contractAddress);
+
+    await getSlotsData(provider, contractAddress, 5);
+    console.log(`Selected slot: ${adminAddressSlot}`);
+    console.log(`Current admin: ${await adminControlled.admin()}`);
+    console.log(`Trying to upgrade admin address to: ${newAdminAddress}`);
+
+    let signer;
+    // We use non-strict unequality as it also includes undefined, etc
+    if (ledgerKeyPath != null) {
+        signer = new EthLedgerSigner(provider, ledgerKeyPath);
+    } else {
+        signer = new ethers.Wallet(process.env.ETH_PRIVATE_KEY, provider);
+    }
+
+    assert.equal(
+        await signer.getAddress(),
+        await adminControlled.admin(),
+        `The used account is not an admin of contract ${contractAddress}`,
+    );
+
+    const ok = await yesno({
+        question: 'WARRING! this change can\'t be reverted, are you sure that you want to change the admin address?',
+    });
```

Comment:
Please specify the only correct phrase for the approval.

So the question will be: `WARRING! Please change all data. This change can not be reverted, do you confirm that you are aware of this and want to change the admin address?` and the yes value will be only `CONFIRM`.

Reply:
Fixed in b4836fa